src/azure-cli/azure/cli/command_modules/resource/_bicep.py

```diff
@@ -302,9 +296,9 @@ def _get_bicep_installation_path(system):
     raise ValidationError(f'The platform "{system}" is not supported.')
 
 
-def _extract_semver(text):
+def _extract_version(text):
     semver_match = re.search(_semver_pattern, text)
-    return semver_match.group(0) if semver_match else None
+    return semver.VersionInfo.parse(semver_match.group(0)) if semver_match else None
```

Comment:
If `semver` is indeed required, we should consider revive https://github.com/Azure/azure-cli/pull/26523.